include/Graph/Graph.hpp

```diff
@@ -936,7 +937,7 @@ namespace CXXGRAPH
 	}
 
 	template <typename T>
-	unsigned long long Graph<T>::setFind(std::vector<Subset> *subsets, const unsigned long long nodeId) const
+	unsigned long long Graph<T>::setFind(std::map<unsigned long long, Subset> *subsets, const unsigned long long nodeId) const
```

Comment:
@sidml  , I have a couple of questions about this function:
- Is there any reason why we are using std::map instead of unordered_map?  This second one is more efficient (although it costs a bit more memory). I think we are focusing mor in efficiency though.
**SUGGESTION**: use unordered_map unless we need the map to order our elements by key.

- Why are we sending the parameters as pointer instead of as a reference?  I am just imaging that somehow a nullptr arrives there...
Also, if we are not modifying the content of the map, it could be passed as a "const reference", then we know that this function is just looking for an element and not changing anything inside of that map.
**SUGGESTION**: define the function as:
_unsigned long long Graph<T>::setFind(const std::unordered_map<unsigned long long, Subset>& subsets, const unsigned long long nodeId) const_

- How do we know that in the map we pass exists the element "nodeId"? Keep in mind that if it does not exist, the program will crash. I think this is a risky implementation.
**SUGGESTION**: I would use "**at**" command to access and element since we want the map to be constant, and would add a quick check of:
_if (subsets.find(nodeId) == subsets.end()) return WHATEVER_WE_SEND_AS_ERROR;_


Finally say that a couple of this comments apply to  "setUnion" and "containsCycle"

Reply:
ok, I apporve this pull request and then you can open the issue ( or solve it)

Reply:
@AlfredCP 
Nice suggestions. I will open new issues. In case you feel anything else can be improved, please feel free to open new issues :smile: